arangod/RocksDBEngine/RocksDBVectorIndex.h

```diff
@@ -85,9 +85,10 @@ class RocksDBVectorIndex final : public RocksDBIndex {
   }
 
   std::pair<std::vector<VectorIndexLabelId>, std::vector<float>> readBatch(
-      std::vector<float>& inputs, RocksDBMethods* rocksDBMethods,
-      transaction::Methods* trx, std::shared_ptr<LogicalCollection> collection,
-      std::size_t count, std::size_t topK);
+      std::vector<float>& inputs, std::optional<std::size_t> nProbe,
```

Comment:
Instead of handing over just the `nProbe` parameter I suggest using the nice `SearchParameters` struct. In future there will only be more variables that affect the search behavior.

Reply:
Makes sense will do that!